### PR TITLE
[Gluten] Fix expression ut result is not check

### PR DESCRIPTION
**Pull request by loneylee**
## What changes were proposed in this pull request?

Fix expression ut result is not check.And add check result.

## How was this patch tested?

Test by UT.

### CODE REVIEW

File: gluten-ut/common/src/test/scala/org/apache/spark/sql/GlutenTestsTrait.scala [hunk 1]
```diff
@@ -128,11 +128,10 @@ trait GlutenTestsTrait extends GlutenTestsCommonTrait {
     val expr = resolver.resolveTimeZones(expression)
     assert(expr.resolved)
 
-    val catalystValue = CatalystTypeConverters.convertToCatalyst(expected)
-
     if(canConvertToDataFrame(inputRow)) {
-      glutenCheckExpression(expr, catalystValue, inputRow)
+      glutenCheckExpression(expr, expected, inputRow)
     } else {
+      val catalystValue = CatalystTypeConverters.convertToCatalyst(expected)
```
Comment (rui-mo) [inline gluten-ut/common/src/test/scala/org/apache/spark/sql/GlutenTestsTrait.scala:134]: Does this mean the execution is on vanilla Spark, and the unit test is not actually verified? If so, can we add a warning here?

**Review by rui-mo — COMMENTED** [review]
Thanks for your fix! I tried this PR locally and found it helped us uncover the incorrectness on Velox backend. I can help add the failed cases of Gluten to the ignore list after the CI finishes.

**Comment by rui-mo** [comment]
Hi @loneylee, opened this PR https://github.com/loneylee/gluten/pull/1 on your branch to ignore these failed unit tests on Velox backend. Could you help merge it? Thanks.

**Comment by rui-mo** [comment]
Hi @loneylee, looks the Velox unit test change is not imported after this force push.

**Comment by loneylee** [comment]
> Hi @loneylee, looks the Velox unit test change is not imported after this force push.

Sorry.My misktake. I have already rebase again.

**Comment by rui-mo** [comment]
> > Hi @loneylee, looks the Velox unit test change is not imported after this force push.
> 
> Sorry.My misktake. I have already rebase again. @rui-mo

Thank you. Never mind.

**Comment by loneylee** [comment]
Run Gluten Clickhouse CI

**Comment by loneylee** [comment]
retest this please

**Comment by loneylee** [comment]
Velox ci has some errors. @rui-mo CC

**Review by rui-mo — APPROVED** [review]

**Comment by loneylee** [comment]
Run Gluten Clickhouse CI

**Comment by rui-mo** [comment]
Run Gluten Clickhouse CI

**Review by rui-mo — APPROVED** [review]

**Pull request merged**